firestore/CONTRIBUTING.md

```diff
@@ -0,0 +1,160 @@
+# Overview
+
+This describes how to Firestore CPP SDK works on supported platforms, and how to
+contribute this Firestore CPP SDK.
+
+Please read `README.md` and `CONTRIBUTING.md` from the repository root first.
+
+# Prerequisites
+
+There is no specific prerequisites for Firestore, `README.md` from the root directory
+should have everything you need.
+
+One slight enhancement is to use [https://github.com/pyenv/pyenv][pyenv] to manage
+your python versions, if you work on multiple projects with different python
+requirements.
+
+For CPP SDK to compile, you generally need your `pyenv which python` to point to
+a Python3 installation.
+
+# Architecture
+
+It is easier to work this Firestore CPP SDK by keeping a high level archetecture in mind:
+
+![architecture.png](architecture.png)
```

Comment:
I am not sure where this diagram came from, but instinctively the arrows seem to point in the wrong direction. I looked up UML diagrams, and how IntelliJ does a Dependency Diagrams (also based on UML I think), and they all seem to point towards the root of dependency.

The first question that popped into my mind when I saw this diagram was: Why does Core SDK have knowledge of Swift? Shouldn't Swift just be an agnostic wrapper around Core SDK? Then I realized that my understanding of arrows (which is consistent with UML) is completely backwards to what the diagram uses.

Reply:
Addressed. I am never a fan of UML, but I guess there is value to follow known conventions.